cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringAgentService.java

```diff
@@ -363,6 +363,11 @@ private void publishStartProgram(String runPayload, PeerInfo peerInfo) throws IO
     try (OutputStream os = programDir.append(DistributedProgramRunner.TETHER_CONF_FILE_NAME).getOutputStream()) {
       cConfCopy.writeXml(os);
     }
+    try (OutputStream os = programDir.append(Constants.Security.Authentication.RUNTIME_TOKEN_FILE).getOutputStream()) {
+      if (files.containsKey(Constants.Security.Authentication.RUNTIME_TOKEN_FILE)) {
```

Comment:
seems like this if should happen outside, so we don't create the file when not needed

Reply:
Fixed